site/src/pages/UserSettingsPage/AppearancePage/AppearanceForm.tsx

```diff
@@ -0,0 +1,294 @@
+
+const ThemePreview: FC<ThemePreviewProps> = ({
+  active,
+  className,
+  displayName,
+  theme,
+}) => {
+  const styles = useMemo(
+    () =>
+      ({
+        container: {
+          backgroundColor: theme.palette.background.default,
+          border: `1px solid ${theme.palette.divider}`,
+          width: 220,
+          color: theme.palette.text.primary,
+          borderRadius: 6,
+          overflow: "clip",
+          userSelect: "none",
+        },
+        containerActive: {
+          outline: `2px solid ${theme.experimental.roles.active.outline}`,
+        },
+        page: {
+          backgroundColor: theme.palette.background.default,
+          color: theme.palette.text.primary,
+        },
+        header: {
+          backgroundColor: theme.palette.background.paper,
+          display: "flex",
+          alignItems: "center",
+          justifyContent: "space-between",
+          padding: "6px 10px",
+          marginBottom: 8,
+          borderBottom: `1px solid ${theme.palette.divider}`,
+        },
+        headerLinks: {
+          display: "flex",
+          alignItems: "center",
+          gap: 6,
+        },
+        headerLink: {
+          backgroundColor: theme.palette.text.secondary,
+          height: 6,
+          width: 20,
+          borderRadius: 3,
+        },
+        activeHeaderLink: {
+          backgroundColor: theme.palette.text.primary,
+        },
+        proxy: {
+          backgroundColor: theme.palette.success.light,
+          height: 6,
+          width: 12,
+          borderRadius: 3,
+        },
+        user: {
+          backgroundColor: theme.palette.text.primary,
+          height: 8,
+          width: 8,
+          borderRadius: 4,
+          float: "right",
+        },
+        body: {
+          width: 120,
+          margin: "auto",
+        },
+        title: {
+          backgroundColor: theme.palette.text.primary,
+          height: 8,
+          width: 45,
+          borderRadius: 4,
+          marginBottom: 6,
+        },
+        table: {
+          border: `1px solid ${theme.palette.divider}`,
+          borderBottom: "none",
+          borderTopLeftRadius: 3,
+          borderTopRightRadius: 3,
+          overflow: "clip",
+        },
+        tableHeader: {
+          backgroundColor: theme.palette.background.paper,
+          height: 10,
+          margin: -1,
+        },
+        label: {
+          borderTop: `1px solid ${theme.palette.divider}`,
+          padding: "4px 12px",
+          fontSize: 14,
+        },
+        workspace: {
+          borderTop: `1px solid ${theme.palette.divider}`,
+          height: 15,
+
+          "&::after": {
+            content: '""',
+            display: "block",
+            marginTop: 4,
+            marginLeft: 4,
+            backgroundColor: theme.palette.text.disabled,
+            height: 6,
+            width: 30,
+            borderRadius: 3,
+          },
+        },
+      }) satisfies Record<string, Interpolation<never>>,
```

Comment:
because these styles will be passed to Emotion (hence the need for `Interpolation`) but they shouldn't be allowed to use the theme emotion provides. they should only be allowed access to the theme they are a preview of. specifying `never` would make it a type error to use an `Interpolation` of the form `(theme) => ({ ... })`, because `theme` has no valid instantiation that satisfies the type `never`.